org.lflang/src/org/lflang/TargetProperty.java

```diff
@@ -171,6 +174,9 @@
     FILES("files", UnionType.FILE_OR_FILE_ARRAY, Arrays.asList(Target.ALL),
             (config, value) -> {
                 config.fileNames = ASTUtils.toListOfStrings(value);
+            },
+            (config, value) -> {
+                config.fileNames.addAll(ASTUtils.toListOfStrings(value));
```

Comment:
Not sure this will work as expected. Are file names resolved with respect to the main file or with respect to the imported file?

Reply:
I think the files need to be resolved while constructing the target config for the imported file, meaning the file names should be a list of files or paths, not strings.

Reply:
Yes, I agree that this could potentially be dangerous. The CGenerator does not merge `files` target properties like that.
We currently keep a list of [LFResources](https://github.com/lf-lang/lingua-franca/blob/21a68aa44bf6f0f97746a92d8ee51d37788d7f9a/org.lflang/src/org/lflang/generator/LFResource.java) and copy user files for each resource as we come across reactors that belong to that resource at the time of code generation.